src/parsers/ddi_parser.jl

```diff
@@ -0,0 +1,189 @@
+    vartype::String = ""
+end
+
+@kwdef mutable struct DDIInfo
+    filepath::String
+    conditions::String = ""
+    citation::String = ""
+    ipums_project::String = ""
+    extract_notes::String = ""
+    extract_date::String = ""
+    variable_info::Vector{DDIVariable} = DDIVariable[]
+    _xml_doc::EzXML.Document = EzXML.XMLDocument()
+    _ns::String = ""
+end
+
+
+
+#=
+
+Set of XPaths for variables in DDI (.xml) files
+
+=#
+
+EXTRACT_CONDITIONS = "/x:codeBook/x:stdyDscr/x:dataAccs/x:useStmt/x:conditions"
+EXTRACT_CITATION = "/x:codeBook/x:stdyDscr/x:dataAccs/x:useStmt/x:citReq"
+EXTRACT_IPUMS_PROJECT = "/x:codeBook/x:stdyDscr/x:citation/x:serStmt/x:serName"
+EXTRACT_NOTES = "/x:codeBook/x:stdyDscr/x:notes"
+EXTRACT_DATE = "/x:codeBook/x:stdyDscr/x:citation/x:prodStmt/x:prodDate/@date"
+
+VAR_NAME_XPATH = "/x:codeBook/x:dataDscr/x:var/@name"
+VAR_STARTPOS_XPATH = "/x:codeBook/x:dataDscr/x:var/x:location/@StartPos"
+VAR_ENDPOS_XPATH = "/x:codeBook/x:dataDscr/x:var/x:location/@EndPos"
+VAR_WIDTH_XPATH = "/x:codeBook/x:dataDscr/x:var/x:location/@width"
+VAR_LABL_XPATH = "/x:codeBook/x:dataDscr/x:var/x:labl"
+VAR_TXT_XPATH = "/x:codeBook/x:dataDscr/x:var/x:txt"
+VAR_DCML_XPATH = "/x:codeBook/x:dataDscr/x:var/@dcml"
+VAR_TYPE_XPATH = "/x:codeBook/x:dataDscr/x:var/x:varFormat/@type"
+VAR_INTERVAL_XPATH = "/x:codeBook/x:dataDscr/x:var/@intrvl"
+VAR_CATEGORY_XPATH = "/x:codeBook/x:dataDscr/x:var/x:catgry"
+
+
+
+"""
+    parse_ddi(filepath)
+
+Parses a valid IPUMs DDI XML file and returns a DDIInfo object.
+
+
+# Examples
+```julia-repl
+julia> parse_ddi("test_extract.xml")
+```
+"""
+function parse_ddi(filepath::String)
+
+    # check to make sure file is xml and not zip
+    
+    _check_that_file_is_xml(filepath)
+
+    # check to make sure file exists
+
+    _check_that_file_exists(filepath)
+    
+    # QUESTION: check if file has multiple files -- then have to handle that
+    
+    # check_if_multiple_files_in_extract() # needed or not ?
+    
+    # read xml file and parse extract level metadata
+    
+    ddifile = EzXML.readxml(filepath)
+    ns = EzXML.namespace(ddifile.root)
+    
+    ddi = DDIInfo(filepath=filepath, _xml_doc=ddifile, _ns=ns)
+    #ns = EzXML.namespace(doc.root)
+    _read_ddi_and_parse_extract_level_metadata!(ddi)
+    
+    # get variables metadata from file
+    _get_var_metadata_from_ddi!(ddi)
+    
+    # TODO determine final output format. For now I return the object.    
+
+    return ddi
+
+end
+
+
+function _check_that_file_is_xml(fname::String)
+    
+    extension = fname[findlast(==('.'), fname)+1:end]
+
+    if extension == "xml"
+        return true
+    else
+        ArgumentError("The DDI file should be in XML format.")
+    end
+
+end
+
+
+function _check_that_file_exists(fname::String)
+
+    if isfile(fname)
+        return true
+    else
+        ArgumentError("The specified file does not exist.")
+    end
+end
+
+
+function _read_ddi_and_parse_extract_level_metadata!(ddi::DDIInfo)
```

Comment:
Can you document what this function does?

Reply:
Yep, I added a docstring for this function.